presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupInfo.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.resourceGroups;
+
+import com.facebook.presto.spi.QueryId;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+public class ResourceGroupInfo
+{
+    private final ResourceGroupId id;
+    private final ResourceGroupState state;
+    private final DataSize softMemoryLimit;
+    private final int maxRunningQueries;
+    private final int maxQueuedQueries;
+    private final List<QueryId> aggregatedRunningQueries;
+    private final int aggregatedQueuedQueryCount;
+    private final DataSize memoryUsage;
+    private final List<ResourceGroupInfo> subGroups;
+    private final int eligibleSubGroups;
```

Comment:
I initially implemented both queuedQueries and runningQueries as list, but then figured that it's not very helpful to show the user what other queries are queued on the resource groups. People are mostly interested in their own queued queries, and other queries that are running (hence blocking them).